File: app/src/main/java/com/unmatchedcounter/model/player/PlayerModel.kt
package com.unmatchedcounter.model.player

import android.os.Parcelable
import androidx.annotation.ColorRes
import com.unmatchedcounter.util.LifeSegmentUtils
import kotlinx.parcelize.Parcelize

@Parcelize
data class PlayerModel(
    val id: Int,
    /** Canonical fighter name from [com.unmatchedcounter.UnmatchedCharacters.NAMES]. */
    val characterName: String,
    /**
     * One display name per life pool (same order as [lifeSegments]), from
     * [com.unmatchedcounter.UnmatchedCharacters] for the chosen character.
     */
    val lifeSegmentLabels: List<String> = listOf(""),
    /** Per-pool ceiling (starting life at game begin); current value never exceeds this. */
    val lifeSegmentMaximums: List<Int> = listOf(0),
    val lifeSegments: List<Int> = listOf(0),
    @ColorRes val colorResId: Int = 0,
) : Parcelable {
    init {
        require(lifeSegments.isNotEmpty() && lifeSegments.size <= LifeSegmentUtils.MAX_LIFE_SEGMENTS) {
            "lifeSegments must have 1..${LifeSegmentUtils.MAX_LIFE_SEGMENTS} values"
        }
        require(lifeSegmentLabels.size == lifeSegments.size) {
            "lifeSegmentLabels must have the same size as lifeSegments"
        }
        require(lifeSegmentMaximums.size == lifeSegments.size) {
            "lifeSegmentMaximums must match lifeSegments size"
        }
    }
}

File: ios/UnmatchedCounter/Model/PlayerModel.swift
import UIKit

struct PlayerModel: Equatable {
    let id: Int
    let characterName: String
    var lifeSegmentLabels: [String]
    var lifeSegmentMaximums: [Int]
    var lifeSegments: [Int]
    var color: PlayerColor

    init(id: Int,
         characterName: String,
         lifeSegmentLabels: [String] = [""],
         lifeSegmentMaximums: [Int] = [0],
         lifeSegments: [Int] = [0],
         color: PlayerColor = .white) {
        precondition(!lifeSegments.isEmpty && lifeSegments.count <= LifeSegmentLimits.maxLifeSegments,
                     "lifeSegments must have 1..\(LifeSegmentLimits.maxLifeSegments) values")
        precondition(lifeSegmentLabels.count == lifeSegments.count,
                     "lifeSegmentLabels must match lifeSegments size")
        precondition(lifeSegmentMaximums.count == lifeSegments.count,
                     "lifeSegmentMaximums must match lifeSegments size")
        self.id = id
        self.characterName = characterName
        self.lifeSegmentLabels = lifeSegmentLabels
        self.lifeSegmentMaximums = lifeSegmentMaximums
        self.lifeSegments = lifeSegments
        self.color = color
    }
}
